app/src/main/java/com/github/se/eventradar/ui/home/Home.kt

```diff
@@ -132,12 +135,12 @@ fun HomeScreen(
 
     SearchBarAndFilter(
         searchQuery = uiState.searchQuery,
-        onSearchQueryChange = { uiState.searchQuery = it },
+        onSearchQueryChange = { newQuery -> viewModel.onSearchQueryChange(newQuery) },
         onSearchButtonClick = {
           // Handle search button click
         },
         showFilterPopUp = uiState.isFilterDialogOpen,
-        setShowFilterPopUp = { uiState.isFilterDialogOpen = it },
+        setShowFilterPopUp = { viewModel.changeFilterDialogOpen() },
         modifier =
             Modifier.padding(horizontal = 16.dp, vertical = 8.dp)
                 .testTag("searchBarAndFilter")
```

Comment:
small nit: instead of if (getTabIndexFromTabEnum(uiState.tab) == 0), maybe uiState.tab == Tab.Browsing would be more readable